File: python/hw1.py
import os
os.system("cls")

print(str("5") + str(5))
print(int("5") - 5)
print(int("5") * int("2"))
print(int("10") / 2)
print(5 + True)
print(int("10") - True)
print("5" + "true")
print(5 + 0)
print("NaN")
print(200 + 50)
print(str(25) + " years old")
print("true" if ("false" != "") else "false")
print(int("10") - True)

temp = 30
conventioner = (temp * 9.0 / 5.0) + 32
print(conventioner)

# Budget program
totalBudget = float(input("Enter your total budget: "))
expense1 = float(input("Enter expense 1: "))
expense2 = float(input("Enter expense 2: "))
expense3 = float(input("Enter expense 3: "))

totalExpenses = expense1 + expense2 + expense3
remaining = totalBudget - totalExpenses

print("Total budget:", totalBudget)
print("You spent", expense1, "first time")
print("You spent", expense2, "second time")
print("You spent", expense3, "third time")
print("After spending you have:", remaining)

if remaining < 0:
    print("Overspent! You are over budget", file=__import__("sys").stderr)
else:
    print("You're within budget.")

File: c++/hw1.cpp
#include <iostream>
using namespace std;

int main() {
    cout << string("5") + to_string(5) << endl;
    cout << stoi("5") - 5 << endl;
    cout << stoi("5") * stoi("2") << endl;
    cout << stoi("10") / 2 << endl;
    cout << 5 + true << endl;
    cout << stoi("10") - true << endl;
    cout << string("5") + "true" << endl;
    cout << 5 + 0 << endl;
    cout << "NaN" << endl;

    cout << 200 + 50 << endl;
    cout << to_string(25) + " years old" << endl;
    cout << (string("false") != "" ? "true" : "false") << endl;
    cout << stoi("10") - true << endl;

    const double temp = 30;
    double conventioner = (temp * 9.0 / 5.0) + 32;
    cout << conventioner << endl;

    double totalBudget, expense1, expense2, expense3;
    cout << "Enter your total budget: "; cin >> totalBudget;
    cout << "Enter expense 1: "; cin >> expense1;
    cout << "Enter expense 2: "; cin >> expense2;
    cout << "Enter expense 3: "; cin >> expense3;

    double totalExpenses = expense1 + expense2 + expense3;
    double remaining = totalBudget - totalExpenses;

    cout << "Total budget: " << totalBudget << endl;
    cout << "You spent " << expense1 << " first time" << endl;
    cout << "You spent " << expense2 << " second time" << endl;
    cout << "You spent " << expense3 << " third time" << endl;
    cout << "After spending you have: " << remaining << endl;

    if (remaining < 0) {
        cerr << "Overspent! You are over budget" << endl;
    } else {
        cout << "You're within budget." << endl;
    }
    return 0;
}
